lib/validation.js

```diff
@@ -217,6 +217,26 @@ function validateAsyncHeaders (validatePromise, context, request) {
     })
 }
 
+function validateAsyncHooks (name, fn) {
+  if (fn.constructor.name !== 'AsyncFunction') {
+    return true
+  }
+
+  const validateParametersLengthFunction = {
+    onSend: length => !(length === 4),
+    preSerialization: length => !(length === 4),
+    onError: length => !(length === 4),
+    preParsing: length => !(length === 4),
+    onReady: length => !(length !== 0),
+    onListen: length => !(length !== 0),
+    onRequestAbort: length => !(length !== 1)
```

Comment:
Please avoid using negated false. `length => length !== 4` is far easier to read than `length => !(length === 4)`.

Reply:
My bad, I'm fixing